beets/metadata_plugins.py

```diff
@@ -0,0 +1,380 @@
+
+        :param items: List of items in the album
+        :param artist: Album artist
+        :param album: Album name
+        :param va_likely: Whether the album is likely to be by various artists
+        """
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def item_candidates(
+        self, item: Item, artist: str, title: str
+    ) -> Iterable[TrackInfo]:
+        """Return :py:class:`TrackInfo` candidates that match the given track.
+
+        Used in the autotag functionality to search for tracks.
+
+        :param item: Track item
+        :param artist: Track artist
+        :param title: Track title
+        """
+        raise NotImplementedError
+
+    def albums_for_ids(self, ids: Sequence[str]) -> Iterable[AlbumInfo | None]:
+        """Batch lookup of album metadata for a list of album IDs.
+
+        Given a list of album identifiers, yields corresponding AlbumInfo objects.
+        Missing albums result in None values in the output iterator.
+        Plugins may implement this for optimized batched lookups instead of
+        single calls to album_for_id.
+        """
+
+        return (self.album_for_id(id) for id in ids)
+
+    def tracks_for_ids(self, ids: Sequence[str]) -> Iterable[TrackInfo | None]:
+        """Batch lookup of track metadata for a list of track IDs.
+
+        Given a list of track identifiers, yields corresponding TrackInfo objects.
+        Missing tracks result in None values in the output iterator.
+        Plugins may implement this for optimized batched lookups instead of
+        single calls to track_for_id.
+        """
+
+        return (self.track_for_id(id) for id in ids)
+
+    def album_distance(
+        self,
+        items: Sequence[Item],
+        album_info: AlbumInfo,
+        mapping: dict[Item, TrackInfo],
+    ) -> Distance:
+        """Calculate the distance for an album based on its items and album info."""
+        return _get_distance(
+            data_source=self.data_source, info=album_info, config=self.config
+        )
+
+    def track_distance(
+        self,
+        item: Item,
+        info: TrackInfo,
+    ) -> Distance:
+        """Calculate the distance for a track based on its item and track info."""
+        return _get_distance(
+            data_source=self.data_source, info=info, config=self.config
+        )
+
+    @cached_property
+    def data_source(self) -> str:
+        """The data source name for this plugin.
+
+        This is inferred from the plugin name.
+        """
+        return self.__class__.__name__.replace("Plugin", "")
+
+    def extract_release_id(self, url: str) -> str | None:
+        """Extract an ID from a URL for this metadata source plugin.
+
+        Uses the plugin's data source name to determine the ID format and
+        extracts the ID from a given URL.
+        """
+        return extract_release_id(self.data_source, url)
+
+    @staticmethod
+    def get_artist_str(
```

Comment:
```suggestion
    def get_artist(
```

Let's use the same name as previously - embedding type into method names is an anti-pattern, and we generally don't use this across the codebase.